README.md

```diff
@@ -188,6 +189,14 @@ STS endpoint by setting `AWS_STS_REGIONAL_ENDPOINTS=regional`, because both the
 SDK and Vault IAM auth method default to using the global endpoint in many regions.
 See documentation on [`sts_regional_endpoints`][lambda-sts-regional-endpoints] for more information.
 
+### Caching
+
+Caching can be configured for the extension's local proxy server so that it does not forward every HTTP request to Vault. To turn on caching, set VAULT_CACHE_TTL to a valid value that is parseable by time.Duration in Go, for example, "15m", "1h", "2m3s" or "1h2m3s", depending on application needs. An invalid or negative value will be treated the same as a missing value, in which case, caching will not be set up and enabled.
+
+Only requests with HTTP method of "GET", and a query parameter of "cacheable" with value of "1" will be cached at the proxy server. Cached value will be returned when the same request is made, without making another call to Vault. Currently, the cache key is composed of the request's URL path and its "version" query parameter value. The main consideration behind caching design is to make caching an explicit opt-in at the request level, so that it is only enabled for scenarios where caching makes sense without negative impact over others.
```

Comment:
Have you considered using a header instead of query parameters for controlling the caching behavior? For example, if we had a `X-Vault-Cache-Control` header, it could be used like this:

```
cacheable=1           --> X-Vault-Cache-Control: cache
cacheable=1,recache=1 --> X-Vault-Cache-Control: recache
```

It seems like that might simplify the UX a little, and leave room for future cache options to be sent from the client without adding more parameters.

Reply:
Like the idea - and it is quite easy for clients to add a header like such.